spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/admin/SpannerDatabaseAdminTemplate.java

```diff
@@ -88,19 +88,19 @@ public SpannerDatabaseAdminTemplate(DatabaseAdminClient databaseAdminClient,
 	 * already exist.
 	 */
 	public void executeDdlStrings(Iterable<String> ddlStrings,
-			boolean createDatabase) {
+			boolean createDatabase) throws InterruptedException {
```

Comment:
Can we just catch the `InterruptedException` and throw the `SpannerDataException`?

Reply:
So, did you try to just re-interrupt in the catch block with `Thread.currentThread().interrupt()`?

Reply:
No, I didn't think that was preferable to just throwing the exception. 

The spanner admin template isn't the place to interrupt the thread in my opinion. It should be just as blind to handling the interrupt as the DDL request it submitted to the client library.

Reply:
I don't think it's very user-friendly. Why don't we just return the future then?

Reply:
I think it's perfectly user-friendly. The user is totally free to ignore the exception because they aren't subject to sonarcube haha.

Returning a future also necessarily means they have to deal with the exception, too. Not to mention it's far far less user friendly. All the user wanted to do was run some DDL strings. Why should he have to worry about  a future??

Reply:
The future would allow them to not block the thread on the operation completion, unless they want to.

Reply:
Yes that's certainly true (and they would need to handle the interrupt too), and they would have even more flexibility if they used the client lib directly. The purpose of this method is purely for convenience, and I like the clean logical flow of DDL string straight to tables being created without worry about intermediate Java things like futures and threads.

Reply:
Sure, I buy into the concept of convenience at the template level, which for me also includes not rethrowing checked exceptions. AFIAK, Spring's convention is no checked exceptions in templates. @artembilan Can you comment on this please?

Reply:
I don't think it's inconvenient for the user to  catch and ignore the exception. These types of admin operations happening over the internet are usually things a user would have a in a try-catch block anyway.